_posts/2024-06-10-infinispan-15-queries.adoc

```diff
@@ -0,0 +1,317 @@
+
+[[rest_projections]]
+== Rest queries with projections
+
+Recently we introduced score and versions projections, that along with entity and field projections are the all
+kinds of projections we support with the Infinispan queries.
+
+The same projections are now available using the REST query API.
+
+For instance if we have an entity like the following:
+
+[source,java]
+----
+@Indexed(index = "play")
+public class Game {
+
+   private final String name;
+   private final String description;
+
+   @ProtoFactory
+   public Game(String name, String description) {
+      this.name = name;
+      this.description = description;
+   }
+
+   @Keyword(projectable = true, sortable = true)
+   @ProtoField(1)
+   public String getName() {
+      return name;
+   }
+
+   @Text
+   @ProtoField(2)
+   public String getDescription() {
+      return description;
+   }
+}
+----
+
+
+Running the following query:
+
+
+[source,sql]
+----
+select g, g.description, version(g), score(g) from Game g where g.description : 'bla3'
+----
+
+
+This is a possible outcome:
+
+
+[source,json]
+----
+{
+    "hit_count": 1,
+    "hit_count_exact": true,
+    "hits": [
+        {
+            "hit": {
+                "version()": 7,
+                "description": "bla bla3",
+                "*": {
+                    "_type": "Game",
+                    "name": "bla3",
+                    "description": "bla bla3"
+                },
+                "score()": 0.90565
+            }
+        }
+    ]
+}
+----
+
+
+In this case we have requested the version projection using the projection function `version(g)`
+and receiving the corresponding result in the `version()` attribute of the returned `hit`.
+Similarly, a `score()` projection is produced.
+The special value `*` corresponds to the entity projection result, in which the special field `_type` corresponds to the type of the entity.
+
+
+[[continuous-queries]]
+== Query cache API
+
+The use of the search factory to access the query APIs is no longer required and it is now deprecated.
+
+From Infinispan 15 we can simply create a query instance from both a remote or an embedded cache (the API is the same)
+using the method `cache#query`.
+Here is an example:
+
+[source,java]
+----
+Query<Person> query = myCache.query("FROM space.Person WHERE name = 'user1' AND age > 20");
+----
+
+From this point forward we will use the usual API to configure and run the query. For instance:
+
+[source,java]
+----
+query.startOffset(10);
+query.maxResults(10);
+QueryResult<Person> = query.execute();
+----
+
+Similarly, it is possible to get a continuous query instance from the same cache (both remote and embedded),
+using the method `cache#continuousQuery`:
+
+[source,java]
+----
+ContinuousQuery<Integer, Person> continuousQuery = myCache.continuousQuery();
+----
+
+From which as usual it will be possible to define the continuous callback:
```

Comment:
I would remove this and group the example

Reply:
I would prefer to keep it, to see where the story ends, even if we didn't changed this API.